rusoto/core/src/signature.rs

```diff
@@ -330,13 +362,15 @@ fn skipped_headers(header: &str) -> bool {
     ["authorization", "content-length", "user-agent"].contains(&header)
 }
 
+/// Returns standardised URI
 fn canonical_uri(path: &str) -> String {
     match path {
         "" => "/".to_string(),
         _ => encode_uri_path(path),
     }
 }
 
+/// Returns standardised query while iterating through the given paramaters
```

Comment:
I'd point to: http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html#query-string-auth-v4-signing for this one. Query strings are lucky enough to get their own page.